astropy/io/ascii/core.py

```diff
@@ -1345,8 +1345,17 @@ def _get_reader(Reader, Inputter=None, Outputter=None, **kwargs):
             kwargs['Inputter'] = Inputter
         return Reader(**kwargs)
 
+    # If user explicitly passed a fast reader with 'force' or with non-default
+    # options for the fast reader, raise an error for slow readers
     if 'fast_reader' in kwargs:
-        del kwargs['fast_reader'] # ignore fast_reader parameter for slow readers
+        if kwargs['fast_reader'] == 'force' or \
+           isinstance(kwargs['fast_reader'] , dict):
+            raise ParameterError('fast_reader required with ' +
```

Comment:
This changes the return of `ascii.read` with `fast_reader` explicitly requested, if the `C` parser fails: from fallback to the slow reader to an exception.